### PR TITLE
Introduces MockLLM and provides an example

### PR DESCRIPTION
**Pull request by delip**
**Rationale:** For testing and other purposes, we need a Log10 LLM instance that mimics an LLM chat call but doesn't rely on existing LLMs, local or otherwise. The NoopLLM class provides an interface that always produces a constant output. However, for testing feedback/logging, you might want to have a mock LLM that produces different but deterministic outputs for different inputs. This PR provides that.

As a choice, the current implementation only provides `.chat()` to facilitate my current work and leaves `.text()` for future implementation.

### CODE REVIEW

**Review by wenzhe-log10 — CHANGES_REQUESTED** [review]
Thanks for contributing. 

High level question - what's the use case for the MockLLM? do we expect the user to set prompt messages and response message directly?

Please also run `ruff check log10 examples` and `ruff format log10 examples` for lint/format. current check failed.

File: log10/llm.py [hunk 1]
```diff
@@ -250,3 +251,64 @@ def chat(self, messages: List[Message], hparams: dict = None) -> ChatCompletion:
...
+        If mock_function is not provided, it is assigned to an identity function.
+        '''
+        super().__init__(hparams, log10_config)
+        self.mock_function = mock_function if mock_function else lambda x: x
+
+    def chat(self, messages: List[Message], hparams: dict = None) -> ChatCompletion:
+        request = self.chat_request(messages, hparams)
+
+        start_time = time.perf_counter()
+        content = self.mock_function(messages[-1].content)
```
[the remark this answers is not in the document]
Comment (delip) [reply log10/llm.py:276]: @wenzhe-log10 yes, extra check won't hurt. will do. 

Yes, mock_function does that. However, I will update this with another commit that does what you are asking for from a different interface.

File: log10/llm.py [hunk 2]
```diff
@@ -250,3 +251,64 @@ def chat(self, messages: List[Message], hparams: dict = None) -> ChatCompletion:
     def text(self, prompt: str, hparams: dict = None) -> TextCompletion:
         logging.info("Received text completion requst: " + prompt)
         return TextCompletion(text="I'm not a real LLM")
+
+
+class MockLLM(LLM):
+    '''
+    MockLLM is an LLM interface that allows you to mock the behavior of an LLM using any Python function. 
+    It is useful for log10 testing and development without having to setup or call a real LLM.
+    See examples/feedback/echo for an example.
```
[the remark this answers is not in the document]
Comment (delip) [reply log10/llm.py:260]: Sure. I like yours. I will add that.

File: log10/llm.py [hunk 3]
```diff
@@ -250,3 +251,64 @@ def chat(self, messages: List[Message], hparams: dict = None) -> ChatCompletion:
...
+        start_time = time.perf_counter()
+        content = self.mock_function(messages[-1].content)
+
+        self.completion_id = self.log_start(request, Kind.chat)
+        completion = {
+            "choices": [
+                {
+                    "message": {
+                        "role": "assistant",
+                        "content": content,
```
[the remark this answers is not in the document]
Comment (delip) [reply log10/llm.py:284]: please check the `chat_expected` method I will be committing soon.

**Review by wenzhe-log10 — CHANGES_REQUESTED** [review]
cool. looks good. 
Do we still need the two example files in `examples/feedback/echo`? Maybe remove since you have add the doc string now.
Last thing is to update the ruff format (failed [here](https://github.com/log10-io/log10/actions/runs/8665651511/job/23764791278?pr=130)).

**Comment by wenzhe-log10** [comment]
stale PR. closing this for now

**Pull request closed without merging**